dart/lib/src/event_processors/user_event_processor.dart

```diff
@@ -0,0 +1,31 @@
+import 'dart:async';
+import '../protocol.dart';
+import '../sentry_options.dart';
+
+/// Default value for [User.ipAddress]. It gets set when an event does not have
+/// a user and IP address. Only applies if [SentryOptions.sendDefaultPii] is set
+/// to true.
+const _defaultIpAddress = '{{auto}}';
+
+/// This adds an [EventProcessor] which modifies the users IP address according
+/// to [SentryOptions.sendDefaultPii].
+void addUserEventProcessor(SentryOptions options) {
+  options.addEventProcessor(
+      (event, {hint}) => userEventProcessor(options, event, hint: hint));
+}
+
+FutureOr<SentryEvent?> userEventProcessor(
```

Comment:
some folks are against using event processors when not needed, I guess that would be a use case to not use it as the code is Pure Dart.
this could be done for example -> `SentryClient._prepareEvent`